exasol/slc/internal/tasks/build/docker_build.py

```diff
@@ -1,48 +1,65 @@
+# pylint: disable=not-an-iterable
+from typing import Dict, Generator, Optional, Set, Tuple
+
 import luigi
+from exasol_integration_test_docker_environment.lib.base.abstract_task_future import (
+    AbstractTaskFuture,
+)
+from exasol_integration_test_docker_environment.lib.base.base_task import BaseTask
 from exasol_integration_test_docker_environment.lib.base.flavor_task import (
     FlavorsBaseTask,
 )
-from exasol_integration_test_docker_environment.lib.base.json_pickle_target import (
-    JsonPickleTarget,
-)
-from exasol_integration_test_docker_environment.lib.base.pickle_target import (
-    PickleTarget,
+from exasol_integration_test_docker_environment.lib.docker.images.image_info import (
+    ImageInfo,
 )
 from luigi import Config
-from luigi.format import Nop
 
 from exasol.slc.internal.tasks.build.docker_flavor_build_base import (
     DockerFlavorBuildBase,
 )
 
 
 class DockerBuildParameter(Config):
-    goals = luigi.ListParameter()
-    shortcut_build = luigi.BoolParameter(True)
+    goals: Tuple[str, ...] = luigi.ListParameter()  # type: ignore
+    shortcut_build: bool = luigi.BoolParameter(True)  # type: ignore
 
 
 class DockerBuild(FlavorsBaseTask, DockerBuildParameter):
 
-    def __init__(self, *args, **kwargs):
-        self._images_futures = None
+    def __init__(self, *args, **kwargs) -> None:
+        self._images_futures: Optional[AbstractTaskFuture] = None
         super().__init__(*args, **kwargs)
 
-    def register_required(self):
+    def register_required(self) -> None:
         tasks = self.create_tasks_for_flavors_with_common_params(DockerFlavorBuild)
         self._images_futures = self.register_dependencies(tasks)
 
-    def run_task(self):
-        image_info = self.get_values_from_futures(self._images_futures)
-        self.return_object(image_info)
+    def run_task(self) -> None:
+        image_infos: Dict[str, Dict[str, ImageInfo]] = self.get_values_from_futures(
+            self._images_futures
+        )
+        assert isinstance(image_infos, dict)
+        assert all(isinstance(x, str) for x in image_infos.keys())
+        assert all(isinstance(x, dict) for x in image_infos.values())
+        assert all(isinstance(y, str) for x in image_infos.values() for y in x.keys())
+        assert all(
+            isinstance(y, ImageInfo) for x in image_infos.values() for y in x.values()
+        )
```

Comment:
I wanted to be sure that the expected type hints for the deserialized objectes are really correcct. Haven't found a better way to check the type of a `Dict`....